app/helpers/moments_helper.rb

```diff
@@ -36,11 +61,88 @@ def moments_stats
     end
     result + '</div>'
   end
+
+  def setup_moment_or_strategy(element)
+    if element.class.name == 'Moment'
+      return {
+        url_helper: moment_path(element),
+        link: edit_moment_path(element),
+        moods: element.mood.map { |item| Mood.find(item).name },
```

Comment:
I think this can be reduced to
```suggestion
        moods: Mood.where(id: element.mood).map(&:name),
```
or 
```suggestion
        moods: Mood.find(element.mood).map(&:name),
```
but I would probably have it in the models